README.md

```diff
@@ -11,7 +11,77 @@ WasmSDK is a toolchain for WebAssembly (WASM).  In addition to being a general p
 - _eosio-cpp_, _eosio-cc_, _eosio-ld_, and _eosio-pp_ are set the core set of tools that you will interact with.
     * These are the C++ compiler, C compiler, linker and postpass tools.
 - A simple CMake interface to build EOSIO smart contracts against WasmSDK
-- ABI generator (Coming Soon)
+- ABI generator
+
+# How to use eosio-abigen
+## using with eosio-cpp
+To generate an abi with ```eosio-cpp```, the only flag you need to pass to ```eosio-cpp``` is `-abigen`, this will tell the compiler to run `eosio-abigen` after compilation and linking stages.  If the output filename is specified as a '.wasm' file with the `-o` option (e.g. \<filename\>.wasm) then eosio-cpp with tell the abi generator to create the abi with the name \<filename\>.abi, if no '.wasm' suffix is used then the resulting output filename is still \<filename\>.abi 
+
+Example:
+```bash
+$ eosio-cpp hello.cpp -o hello.wasm -abigen
+```
+This will generate two files:
+* The compiled binary wasm (hello.wasm)
+* The generated abi file (hello.abi)
+
+## using eosio-abigen alone
+To generate an abi with ```eosio-abigen```, only requires that you give the main '.cpp' file to compile and the output filename `-output`.
+
+Example:
+```bash
+$ eosio-abigen hello.cpp -output=hello.abi
```

Comment:
Why -output vs. the "usual" -o?

Reply:
-o can also be passed to eosio-abigen as part of extra C++ options (after '--' ), and is ignored if packed with them.  So, I didn't want any confusion.

Reply:
It's clang libtooling's option parser, so the option is -output or --output, both are interchangeable.  I used the convention of clang/gcc i.e. one dash for the example text, but if you think that would be confusing we could use eosio's convention.